bitbots_vision/bitbots_vision/yoeo_vision.py

```diff
@@ -0,0 +1,257 @@
+                os.path.exists(os.path.join(model_path, "yoeo", "mod.json"))
+        return exists
+
+    def _set_up_yoeo_handler(self, new_config: Dict) -> None:
+        if self._new_yoeo_handler_is_needed(new_config):
+            self._instantiate_new_yoeo_handler(new_config)
+        elif self._yoeo_parameters_have_changed(new_config):
+            self._yoeo_handler.set_config(new_config)
+
+    def _new_yoeo_handler_is_needed(self, new_config: Dict) -> bool:
+        return self._yoeo_handler is None or \
+               ros_utils.config_param_change(self._config, new_config, ['neural_network_type'])
+
+    def _instantiate_new_yoeo_handler(self, new_config: Dict) -> None:
+        neural_network_type = new_config["neural_network_type"]
+        model_path = self._get_model_path(new_config)
+        if neural_network_type == "openvino":
+            self._yoeo_handler = yoeo_handler.YOEOHandlerOpenVino(new_config, model_path)
+        elif neural_network_type == "onnx":
+            self._yoeo_handler = yoeo_handler.YOEOHandlerONNX(new_config, model_path)
+        elif neural_network_type == "pytorch":
+            self._yoeo_handler = yoeo_handler.YOEOHandlerPytorch(new_config, model_path)
+        elif neural_network_type == "tvm":
+            self._yoeo_handler = yoeo_handler.YOEOHandlerTVM(new_config, model_path)
+        logger.info(f"Using {self._yoeo_handler.__class__.__name__}")
+
+    def _yoeo_parameters_have_changed(self, new_config: Dict) -> bool:
+        return ros_utils.config_param_change(self._config, new_config, r'yoeo_')
+
+    def _set_up_vision_components(self, new_config: Dict) -> None:
+        self._vision_components = []
+
+        if new_config["component_camera_cap_check_active"]:
+            self._vision_components.append(CameraCapCheckComponent(self))
+        if new_config["component_ball_detection_active"]:
+            self._vision_components.append(YOEOBallDetectionComponent(self))
+        if new_config["component_obstacle_detection_active"]:
+            self._vision_components.append(YOEOObstacleDetectionComponent(self))
+        if new_config["component_goalpost_detection_active"]:
+            self._vision_components.append(YOEOGoalpostDetectionComponent(self))
+        if new_config["component_line_detection_active"]:
+            self._vision_components.append(YOEOLineDetectionComponent(self))
+        if new_config["component_field_boundary_detection_active"]:
+            self._vision_components.append(YOEOFieldBoundaryDetectionComponent(self))
+        if new_config["component_field_detection_active"]:
+            self._vision_components.append(YOEOFieldDetectionComponent(self))
+        if new_config["component_debug_image_active"]:
+            self._vision_components.append(DebugImageComponent(self))
+
+        for vision_component in self._vision_components:
+            vision_component.configure(new_config, self._yoeo_handler)
```

Comment:
Reminder: Why does every vision_component need a yoeo_handler?

Reply:
Not every component does, but most do. Those which do not need it do not use it in their configure methods. There just has to be a common interface...